tests/impl/uJIT-tests-Lua/suite/string.t

```diff
@@ -107,3 +107,8 @@ $tester->run('bufstr-const-fold.lua', args => '-p-')
     ->stdout_has_no(qr/TRACE.+?abort.+?/)
     ->stdout_has_no(q/BUFSTR/)  # all BUFSTRs were folded away
 ;
+
+$tester->run('find-fold-bug.lua', args => '-p-')
+    ->exit_ok
+    ->stdout_has_no(qr/TRACE.+?abort.?/)
```

Comment:
Was the trace abort the only observable behaviour of this bug? I think there needs to be a check that fold transformations are not buggy (`std_out_has_no(/* before fix output */)`)

Reply:
I guess it is enough. Previous implementation of `string.find` recording breaks the sematics while folding. Thereby we need to check whether a trace is assembled and the result is valid (see the assert in the corresponding lua chunk).

Reply:
Was there a trace abort before the change?

Reply:
There were no aborts before (please see the original issue and commits for more info).